src/sentry/models/monitor.py

```diff
@@ -147,6 +148,20 @@ class Meta:
 
     __repr__ = sane_repr("guid", "project_id", "name")
 
+    def save(self, *args, **kwargs):
+        # TODO(epurkhsier): This logic is to be removed when the `guid` field
+        # is removed and a slug is required when creating monitors
+
+        # NOTE: We ONLY set a slug while saving when creating a new monitor and
+        # the slug has not been set. Otherwise existing monitors without slugs
+        # would have their guids changed
+        if self._state.adding is True and self.slug is None:
+            self.guid = uuid4()
```

Comment:
we have to generate a `guid` here because `auto_add` doesn't create it until later, right?

Reply:
Yes correct, unfortunately that happens in the super save call.